### ISSUE TITLE
WS2001 crashes with 'dict_node' object has no attribute 'strip' (reported as E0002)

### ISSUE THREAD

**Issue by neboduus** [aws-cloudformation/cfn-lint#4373]
### CloudFormation Lint Version

cfn-lint 1.44.0

### What operating system are you using?

Linux (GitLab CI Kubernetes executor, container image ci/cfn-lint:latest)

### Describe the bug

When linting a SAM template that includes `Metadata.cfn-lint.config.ignore_checks`, rule **WS2001** raises an exception. cfn-lint reports **E0002** (“Unknown exception while processing rule WS2001”) with:

`'dict_node' object has no attribute 'strip'`

**Steps to reproduce**

1. Install cfn-lint 1.44.0.
2. Save the following as `template.yaml`.
3. pip install cfn_lint_serverless
4. Run: `cfn-lint template.yaml -a cfn_lint_serverless.rules`
5. Observe E0002 with the message above.
**Actual behavior**

WS2001 raises an `AttributeError` because code calls `.strip()` on a value that is a `dict_node` (or similar non-string). That exception is reported as E0002.

**Root cause (analysis)**

WS2001 appears to assume a value is a string and calls `.strip()` on it. For this template (e.g. when processing Metadata or certain nodes), that value can be a `dict_node` (or dict-like YAML node), which has no `.strip()` method.

**Note**

WS2001 is listed in `ignore_checks`, but the rule still runs and crashes; E0002 is still reported.

**Suggested fix**

In WS2001, only call `.strip()` after ensuring the value is a string, or handle dict/node types so `.strip()` is never called on them.

### Expected behavior

**Expected behavior**

WS2001 either reports a finding or does nothing; no uncaught exception and no E0002.

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Description: 'Create resources for configuration api'
Metadata:
    cfn-lint:
        config:
            ignore_checks:
                - WS2001
Parameters:
  Env:
    Type: String
    Default: dev
Transform: AWS::Serverless-2016-10-31
Resources:
  DummyFunction:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: nodejs20.x
      CodeUri: .
```
